GG/GG/TextControl.h

```diff
@@ -165,6 +171,20 @@ class GG_API TextControl : public Control
         the newly rendered text occupies. */
     virtual void SetText(const std::string& str);
 
+    /** Sets the text displayed in this control to \a str \p text_elements
+        pair.  This is faster than SetText without \p text_elements. May resize
+        the window.  If the control was constructed with FORMAT_NOWRAP, calls
+        to this function cause the window to be resized to whatever space the
+        newly rendered text occupies. */
+    virtual void SetText(const std::string& str,
```

Comment:
Does something weird happen if the contents of text_elements is inconsistent with str?